### PR TITLE
Update system physical constants after reassembly

### PR DESCRIPTION
**Pull request by oliverlee**
Steer encoder index position and upper assembly virtual inertia need to be updated after the steering assembly was disassembled and reassembled.
- [x] update encoder index position
- [x] add project for determining virtual inertia
- [x] update upper assembly virtual inertia
- [x] merge #118

### CODE REVIEW

**Comment by oliverlee** [comment]
This isn't ready to be merged, we still need to update the upper assembly virtual inertia however the output of the motor drive appears to be disabled.

File: scripts/plot_sim.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 from load_sim import load_messages, get_records_from_messages, get_time_vector
 
 MAX_KOLLMORGEN_TORQUE = 10.78
-KOLLMORGEN_DAC_ZERO_OFFSET = 2048 - 125
+KOLLMORGEN_DAC_ZERO_OFFSET = 2048 - 175
```
Comment (oliverlee) [inline scripts/plot_sim.py]: -175 is too much

Comment (oliverlee) [inline scripts/plot_sim.py]: -155 seems good

**Comment by oliverlee** [comment]
We should add a project that hardcodes input steer torque to zero for use in determining the upper assembly virtual inertia.

File: projects/flimnap/main.cc [hunk 2]
```diff
@@ -282,13 +284,19 @@ int main(void) {
...
                 -sa::REAR_WHEEL_RADIUS*(angle::encoder_rate(encoder_rear_wheel)));
+        (void)motor_torque; /* not currently used */
 
         /* yaw angle, just use previous state value */
         const float yaw_angle = angle::wrap(bicycle.pose().yaw);
 
         /* calculate rider applied torque */
+#if defined(FLIMNAP_ZERO_INPUT)
+        const float steer_torque = 0.0f;
+        (void)kister_torque;
```
Comment (oliverlee) [inline projects/flimnap/main.cc]: typo, kistler
not sure why this wasn't caught by travis

**Comment by oliverlee** [comment]
```python
In [4]: run ./calculate_handlebar_inertia.py ../tools/build/zero_input.pb
Calculated handlebar inertia is: 0.13847208756545076 kg-m^2
```

**Comment by oliverlee** [comment]
This PR should probably be dependent on #114 or the other way around.

**Comment by oliverlee** [comment]
This is not building in Travis for some reason

**Comment by oliverlee** [comment]
Looks like build jobs aren't sent to Travis if there are merge conflicts.

File: projects/CMakeLists.txt [hunk 3]
```diff
@@ -73,16 +73,20 @@ option(PHOBOS_BUILD_PROJECT_FLIMNAP_WHIPPLE
...
+option(PHOBOS_BUILD_PROJECT_FLIMNAP_ZERO_INPUT
+    "Build Flimnap (static simulator) demo using the Whipple bicycle model with input fixed to zero" TRUE)
+if(PHOBOS_BUILD_PROJECT_FLIMNAP_WHIPPLE OR
+   PHOBOS_BUILD_PROJECT_FLIMNAP_KINEMATIC OR
+   PHOBOS_BUILD_PROJECT_FLIMNAP_ZERO_INPUT)
     add_subdirectory(flimnap)
 endif()
 
-option(PHOBOS_BUILD_PROJECT_GULLIVER "Build New (static simulator) demo" TRUE)
+option(PHOBOS_BUILD_PROJECT_GULLIVER "Build Gulliver (static simulator) demo" TRUE)
```
Comment (mickvangelderen) [inline projects/CMakeLists.txt]: These fixed don't seem to belong in this commit but w/e.

File: projects/flimnap/main.cc [hunk 4]
```diff
@@ -282,13 +284,19 @@ int main(void) {
         const float rear_wheel_angle = -angle::encoder_count<float>(encoder_rear_wheel);
         const float v = velocity_filter.output(
                 -sa::REAR_WHEEL_RADIUS*(angle::encoder_rate(encoder_rear_wheel)));
+        (void)motor_torque; /* not currently used */
 
         /* yaw angle, just use previous state value */
```
Comment (mickvangelderen) [inline projects/flimnap/main.cc]: Comment style

Comment (mickvangelderen) [inline projects/flimnap/main.cc]: Don't fix everything in the file, just the new lines you are committing.

File: projects/inc/saconfig.h [hunk 5]
```diff
@@ -58,6 +58,6 @@ constexpr float STEER_ASSEMBLY_INERTIA = STEER_ASSEMBLY_INERTIA_WITH_WEIGHT; /*
 
 /* moment of inertia of the handlebars and steering column above the torque sensor, about the steer axis 
  * determined from scripts/calculate_handlebar_inertia.py */
-constexpr float HANDLEBAR_INERTIA = 0.118; /* kg-m^2 */
+constexpr float HANDLEBAR_INERTIA = 0.1384; /* kg-m^2 */
```
Comment (mickvangelderen) [inline projects/inc/saconfig.h]: post line comment style

File: projects/CMakeLists.txt (hunk 3 above)
Comment (oliverlee) [inline projects/CMakeLists.txt]: I was being lazy

Comment (mickvangelderen) [inline projects/CMakeLists.txt]: I get that.

File: projects/inc/saconfig.h [hunk 6]
```diff
@@ -51,13 +51,13 @@ constexpr DACConfig dac1cfg1 = {
 };
 constexpr const DACConfig* KOLLM_DAC_CFG = &dac1cfg1;
 
-/* moment of inertia of steering assembly about the steer axis, kg-m^2 */
-constexpr float STEER_ASSEMBLY_INERTIA_WITH_WEIGHT = 0.1942f; /* with 1 kg weight plate on each side of cross bar*/
-constexpr float STEER_ASSEMBLY_INERTIA_WITHOUT_WEIGHT = 0.0828f; /* without weight plates */
-constexpr float STEER_ASSEMBLY_INERTIA = STEER_ASSEMBLY_INERTIA_WITH_WEIGHT; /* default configuration */
+/* Moment of inertia of steering assembly about the steer axis, kg-m^2 */
+constexpr float STEER_ASSEMBLY_INERTIA_WITH_WEIGHT = 0.1942f; // with 1 kg weight plate on each side of cross bar
```
Comment (oliverlee) [inline projects/inc/saconfig.h]: https://github.com/oliverlee/phobos/blob/2dba991be83c8153f664a57f89b4b6faa0cacdda/projects/inc/simbicycle.h#L49

although this gets changed a bit in a pending PR
